src/Tinfoil/Random.hs

```diff
@@ -80,3 +82,10 @@ drawOnce as = do
     n = NE.length as
 
     discard = dropMS extraBits
+
+-- | Generate identifier unique to the currently-running kernel.
+--
+-- https://github.com/torvalds/linux/blob/22763c5cf3690a681551162c15d34d935308c8d7/drivers/char/random.c
+linuxBootId :: IO BootId
+linuxBootId =
+  fmap (BootId . T.encodeUtf8) $ T.readFile "/proc/sys/kernel/random/boot_id"
```

Comment:
Maybe chat about this one, maybe the linoox hate in me coming out, but I don't like this being linux specific. I know there is a little bit of linux specific code around the place (mountain), but for the most part all of our code should work on os-x given that is where a decent proportion of the development is done.